ruby/ql/lib/codeql/ruby/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -176,6 +176,25 @@ API::Node getExtraSuccessorFromInvoke(InvokeNode node, AccessPathToken token) {
   )
 }
 
+pragma[inline]
+API::Node getAFuzzySuccessor(API::Node node) {
+  result = node.getAMember()
+  or
+  result = node.getMethod(_)
+  or
+  result =
+    node.getArgumentAtPosition(any(DataFlowDispatch::ArgumentPosition apos | not apos.isSelf()))
+  or
+  result =
+    node.getParameterAtPosition(any(DataFlowDispatch::ParameterPosition ppos | not ppos.isSelf()))
+  or
+  result = node.getReturn()
+  or
+  result = node.getAnElement()
+  or
+  result = node.getInstance()
```

Comment:
Yes, slightly unfortunate and probably not a user experience for beginners.

I'm thinking if we add a dedicated extensible predicate for fuzzy models, it would translate into regular models that use both `FuzzyLib` and `FuzzyLib!` as the initial type. Or perhaps we could add `FuzzyLib!` as a fuzzy successor of `FuzzyLib`.

Reply:
The current behaviour does make sense under the mindset of "`FuzzyLib.new` is just another method", but it was initially unintuitive for me because I was thinking of object instantiation as something separate from a regular class method call.